mmv1/products/cloudquotas/QuotaPreference.yaml

```diff
@@ -0,0 +1,164 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'QuotaPreference'
+description: |
+  QuotaPreference represents the preferred quota configuration specified for a project, folder or organization. There is only one QuotaPreference resource for a quota value targeting a unique set of dimensions.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Cloud Quotas Overview': 'https://cloud.google.com/docs/quotas/overview'
+  api: 'https://cloud.google.com/docs/quotas/reference/rest/v1/projects.locations.quotaPreferences'
+base_url: '{{parent}}/locations/global/quotaPreferences'
+self_link: '{{parent}}/locations/global/quotaPreferences/{{quota_preference_id}}'
+create_url: '{{parent}}/locations/global/quotaPreferences?quotaPreferenceId={{quota_preference_id}}&ignoreSafetyChecks={{ignore_safety_checks}}'
+update_url: '{{parent}}/locations/global/quotaPreferences/{{quota_preference_id}}?allowMissing={{allow_missing}}&validateOnly={{validate_only}}&ignoreSafetyChecks={{ignore_safety_checks}}'
+update_verb: :PATCH
+update_mask: true
+skip_delete: true
+autogen_async: false
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  encoder: templates/terraform/encoders/quota_preference.go.erb
+
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "cloudquotas_quota_preference_basic"
+    primary_resource_id: "preference"
+    skip_test: true
+    vars:
+      quota_preference_id: "compute_googleapis_com-CPUS-per-project_us-central2"
+    test_env_vars:
+      project: :PROJECT_NAME
+
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'parent'
+    immutable: true
+    url_param_only: true
+    description: The parent of the quota preference. Allowed parents are "projects/[project-id / number]" or "folders/[folder-id / number]" or "organizations/[org-id / number]".
+  - !ruby/object:Api::Type::String
+    name: 'quota_preference_id'
+    immutable: true
+    url_param_only: true
+    default_value: ""
```

Comment:
Do you know if this default is needed?

Reply:
Seems like not needed, API will generate a UUID if given an empty id. Removed.